### DISCUSSION TITLE
tutorials/intro-dgs/04-data-fetchers

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#86] [course feedback]
# tutorials/intro-dgs/04-data-fetchers

https://www.apollographql.com/tutorials/intro-dgs/04-data-fetchers

**Comment by PeterInfraBIM** [comment]
Execution failed for task ':SpotifyDemoApplication.main()'.
> Process 'command 'C:\Program Files\Java\jdk-19\bin\java.exe'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.
Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
For more on this, please refer to https://docs.gradle.org/8.4/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
BUILD FAILED in 5s
4 actionable tasks: 4 executed

Trying to restart the server let to a failure and this error message. It seems it has to do with Gradle. Any suggestions?

**Reply by lizhennessy** [reply]
Hi there @PeterInfraBIM ! Typically when I've run into this error, restarting my entire IDE generally clears it up: gives Gradle a chance to rebuild with the right settings. I wonder if running on JDK 19 might be another source of issue; under the [DGS Framework Getting Started Guide](https://netflix.github.io/dgs/getting-started/), they mention JDK 17 as a requirement (perhaps this is intended as a minimum requirement? To be safe, I've always run on version 17.) If neither of these steps fixes the issue, please feel free to link your repository to me here! I'll be happy to take a look.

**Reply by PeterInfraBIM** [reply]
Thank you @lizhennessy for the quick reply. Somehow I must have done something in the area that you mentioned since it works now perfectly.  Regards, Peter Willems (infraBIM.nl)

**Comment by fredgoya** [comment]
I had to manually add the directories containing the generated files to my source paths. Would have been helpful if the tutorial addressed that

**Reply by lizhennessy** [reply]
Thanks for sharing this feedback @fredgoya ! I'd love to make the adjustments to the course so that anyone else who runs into this same problem can fix it. Can you let me know which IDE you were using, and what you did to fix the problem? (The generated directories were always accessible within my IDE, so I wonder if some setting was already checked!) Thanks so much!

**Reply by fredgoya** [reply]
I'm using VSCode, and I had to add
```
sourceSets {
	main {
		java {
			srcDirs("build/generated-sources/dgs-codegen")
		}
	}
}
```
to `build.gradle.kts`. I'm not a Gradle expert by any means, so there might have been a better way to do it, but that's what made it work for me.

**Reply by lizhennessy** [reply]
Interesting! Thank you for sharing, @fredgoya . I wonder if this is something IntelliJ does automatically out of the box; in any case, this is a great help for others using VS Code. I am going to try it myself! Have a great day!
